src/main/java/core/mate/academy/service/ExcavatorProducer.java

```diff
@@ -0,0 +1,51 @@
+package core.mate.academy.service;
+
+import core.mate.academy.constant.Color;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.util.RandomValue;
+import java.util.ArrayList;
+import java.util.List;
+
+public class ExcavatorProducer implements MachineProducer<Excavator> {
+    private RandomValue randomValue;
+    private Excavator greenExcavator;
+    private Excavator yellowExcavator;
+    private Excavator redExcavator;
```

Comment:
I think this field are not related to the state of the object. So, much better decision it`s create method, like madeDefaultYellowTruk(), madeDefaultRedTruck(), etc. And make this method`s return created objects, and add this object`s to list from those method`s, not from fileds

Reply:
And this help you do make init() method less)